core/src/main/java/io/aiven/klaw/service/AclControllerService.java

```diff
@@ -87,17 +94,18 @@ public ApiResponse createAcl(AclRequestsModel aclRequestsModel) throws KlawExcep
           .build();
     }
 
+    String kafkaFlavor =
+        manageDatabase
+            .getClusters(KafkaClustersType.KAFKA, tenantId)
+            .get(getEnvDetails(aclRequestsModel.getEnvironment(), tenantId).getClusterId())
+            .getKafkaFlavor();
+
```

Comment:
I think there is a potential for an NPE on .getKafkaFlavor

Reply:
I dont think so. Every kafka cluster must have a flavor defined.